x-pack/plugins/file_data_visualizer/public/lazy_load_bundle/component_wrapper.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+
+export const FileDataVisualizerWrapper: FC = () => {
+  const [FileDataVisualizerComponent, setFileDataVisualizerComponent] = useState<FC<{}> | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (FileDataVisualizerComponent === null) {
+      import('../application/file_datavisualizer').then(({ FileDataVisualizer }) => {
+        setFileDataVisualizerComponent(FileDataVisualizer);
```

Comment:
NIT: could use `react.lazy` instead of a manual effect.

Reply:
good idea, that is much nicer.

Reply:
Updated in https://github.com/elastic/kibana/pull/100863/commits/b532118bd2a9a35510b3a5ca1389498de89f1fdd